libraries/llvm/rts.ll

```diff
@@ -442,58 +410,94 @@ define private %Stack @reset(%Stack %oldStack) {
     %size = ptrtoint ptr getelementptr (%StackValue, ptr null, i64 1) to i64
     %stack = call ptr @malloc(i64 %size)
 
+
     %stackMemory = call %Memory @newMemory()
 
-    %stack.0 = insertvalue %StackValue undef, %ReferenceCount 0, 0
-    %stack.1 = insertvalue %StackValue %stack.0, %Memory %stackMemory, 1
-    %stack.2 = insertvalue %StackValue %stack.1, %Region zeroinitializer, 2
-    %stack.3 = insertvalue %StackValue %stack.2, %Prompt %prompt, 3
-    %stack.4 = insertvalue %StackValue %stack.3, %Stack %oldStack, 4
+    %stack.0 = insertvalue %StackValue zeroinitializer, %Memory %stackMemory, 1
+    %stack.1 = insertvalue %StackValue %stack.0, %Prompt %prompt, 3
+    %stack.2 = insertvalue %StackValue %stack.1, %Stack %oldStack, 5
 
-    store %StackValue %stack.4, %Stack %stack
+    store %StackValue %stack.2, %Stack %stack
 
-    ret %Stack %stack
-}
+    %stack_pointer = getelementptr %PromptValue, %Prompt %prompt, i64 0, i32 1
+    store %Stack %stack, ptr %stack_pointer
 
-define private %Stack @resume(%Stack %stack, %Stack %oldStack) alwaysinline {
-    %uniqueStack = call %Stack @uniqueStack(%Stack %stack)
-    tail call void @resumeUnique(%Stack %uniqueStack, %Stack %oldStack)
-    ret %Stack %uniqueStack
+    ret %Stack %stack
 }
 
-define private void @resumeUnique(%Stack %stack, %Stack %oldStack) {
-    %stackRest = getelementptr %StackValue, %Stack %stack, i64 0, i32 4
-    %rest = load %Stack, ptr %stackRest
-    %isNull = icmp eq %Stack %rest, null
-    br i1 %isNull, label %done, label %next
+define private void @updatePrompts(%Stack %stack) alwaysinline {
```

Comment:
Is this `alwaysinline` here important?

Reply:
I think it doesn't matter since it is only used in `resume`, right?

Reply:
Yes, it doesn't really matter.